libnmstate/nm/sriov.py

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.error import NmstateNotImplementedError
+from libnmstate.schema import SRIOV
+
+
+def create_setting(iface_state):
+    setting = iface_state.get(SRIOV.CONFIG_SUBTREE)
+    if setting:
+        raise NmstateNotImplementedError
+
+    return setting
```

Comment:
While the code works as expected, using the variable `setting` for both the config and the return value is a little bit misleading because it would have to return some NM object and not a a changed config subtree. Therefore `return None` would be better and maybe even using `config` instead of `setting` for the variable that contains the config. Or just avoid the variables,

```
if iface_state.get(SRIOV.CONFIG_SUBTREE):
    raise NmstateNotImplementedError

return None
```

Ideally you would also add a small integration tests that ensures that `create_setting()` raises the exception when the config is used in  state.